File: 17-Chain-of-Responsibility/src/go/director.go
package main

import "fmt"

type Director struct {
	BaseApprover
}

func (d *Director) Handle(request PurchaseRequest) {
	if request.Amount <= 20000 {
		fmt.Printf("  Director approved $%.2f for %s\n", request.Amount, request.Purpose)
	} else {
		d.passToNext(request)
	}
}

File: 17-Chain-of-Responsibility/src/java/chapter17/Director.java
package chapter17;

public class Director extends Approver {
    public Director() {
        super("Director", 20000);
    }

    @Override
    protected void approve(PurchaseRequest request) {
        System.out.printf("  %s approved $%.2f for %s%n",
                name, request.getAmount(), request.getPurpose());
    }
}
